flowapi/flowapi/user_model.py

```diff
@@ -88,24 +93,39 @@ def _get_query_kind_and_aggregation_unit(self, query_json: dict) -> Tuple[str, s
 
         Returns
         -------
-        str, str
-            Query kind and aggregation unit
+        list of tuples str, str
+            List of tuples giving query kind and aggregation unit
 
         """
 
         try:
             query_kind = query_json["query_kind"]
+        except KeyError:
+            raise MissingQueryKindError
+        try:
             if query_kind == "spatial_aggregate":
-                return self._get_query_kind_and_aggregation_unit(
+                return self._get_query_kinds_and_aggregation_units(
                     query_json=query_json["locations"]
                 )
-        except KeyError:
-            raise MissingQueryKindError
+            elif query_kind == "joined_spatial_aggregate":
+                location_spec = self._get_query_kinds_and_aggregation_units(
+                    query_json=query_json["locations"]
+                )[0]
+                # Because metric specs don't have an aggregation unit, we stitch on the one
```

Comment:
Similar to my comment on PR #609, the fact that we need to "stitch on" an aggregation unit suggests that the joined spatial aggregate should really accept this parameter itself. (But again, let's leave this for a future PR.)